utils/authorization.go

```diff
@@ -260,6 +276,27 @@ func SetRolePermissionsFromConfig(roles map[string]*model.Role, cfg *model.Confi
 		)
 	}
 
+	// Grant permissions for editing posts.
+	if isLicensed {
+		switch *cfg.ServiceSettings.AllowEditPost {
+		case model.ALLOW_EDIT_POST_NEVER:
+			index := indexInSlice(model.PERMISSION_EDIT_POST.Id, roles[model.SYSTEM_ADMIN_ROLE_ID].Permissions)
```

Comment:
Rather than removing permissions here, I think it might be tidier to do it the same way as the other ones above, so remove it from the "set defaults" function and then add it in the appropriate place here  depending on the license and config value?

Reply:
Actually, https://github.com/mattermost/mattermost-server/pull/8208 has this bit in it too, so we can merge that first and save you the bother of changing it here.

Reply:
@grundleborg Makes no difference to me if we remove here or remove from default. https://github.com/mattermost/mattermost-server/pull/8208 [is removing it from the defaults for channel user](https://github.com/mattermost/mattermost-server/pull/8208/files#diff-6a93e94fc8adb70b1a7468d9e2324560L147), however, it's [adding it to system_user](https://github.com/mattermost/mattermost-server/pull/8208/files#diff-6a93e94fc8adb70b1a7468d9e2324560R283). So is your proposal that we remove it from the system user default and the channel_user default?

Reply:
Just a note that upon closer inspection that PR does not add edit_post to system_user it's just changing the position of that permission in the slice literal.

Reply:
My proposal would be to take the way it's done in #8208 as I think it's more consistent with how we do it for the other config flags, rather than having it in the `channel_user` defaults but then removing it in this function.

Reply:
I should have been more clear that I understood that part. I'm not clear on whether the same should be done for `system_admin` in your proposal because it doesn't seem to be happening in https://github.com/mattermost/mattermost-server/pull/8208.

Reply:
Ah right, understood. I see you've asked the question in #8208 - I'll wait to see JE's explanation there as I'm also now wondering the same thing.

Reply:
I just wish I had some easy way to get to PR 8208, some link or something.

Reply:
Looks like it has been taken care of. Once 8208 is merged I will ensure the changes are reflected here correctly so that we're not removing the permissions in the migration.